src/libostree/ostree-sysroot-deploy.c

```diff
@@ -4213,14 +4253,35 @@ ostree_sysroot_deployment_set_kargs_in_place (OstreeSysroot *self, OstreeDeploym
       ostree_bootconfig_parser_set (new_bootconfig, "options", kargs_str);
 
       g_autofree char *bootconf_name
-          = bootloader_entry_filename (self, self->deployments->len, deployment);
+          = bootloader_entry_name (self, self->deployments->len, deployment);
 
       g_autofree char *bootconfdir = g_strdup_printf ("loader.%d/entries", self->bootversion);
       glnx_autofd int bootconf_dfd = -1;
       if (!glnx_opendirat (self->boot_fd, bootconfdir, TRUE, &bootconf_dfd, error))
         return FALSE;
 
-      if (!ostree_bootconfig_parser_write_at (new_bootconfig, bootconf_dfd, bootconf_name,
+      g_autofree char *bootconf_filename;
+      if (bootloader_is_boot_count_enabled(self, cancellable, error))
+        {
+          guint max_tries = bootloader_get_max_boot_tries (self, cancellable, error);
+
+          if (!ostree_bootconfig_parser_is_parsed (new_bootconfig))
```

Comment:
Just realized that it makes sense to put that into a separate function to avoid code duplication

Reply:
fixed